DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,400 @@
+            else if (CheckAnyAndNotNull(GuildCommandsCreateList) && CheckAnyAndNotNull(GuildCommandsOverwriteList))
+            {
+                foreach (var cmd in GuildCommandsCreateList)
+                {
+                    var discord_backend_command = await ApplicationCommandsExtension._client.CreateGuildApplicationCommandAsync(GuildId, cmd);
+                    Commands.Add(discord_backend_command);
+                }
+
+                foreach (var cmd in GuildCommandsOverwriteList)
+                {
+                    var command = cmd.Value;
+                    var discord_backend_command = await ApplicationCommandsExtension._client.EditGuildApplicationCommandAsync(GuildId, cmd.Key, action =>
+                    {
+                        action.Name = command.Name;
+                        action.NameLocalizations = command.NameLocalizations;
+                        action.Description = command.Description;
+                        action.DescriptionLocalizations = command.DescriptionLocalizations;
+                        if(command.Options != null && command.Options.Any())
+                            action.Options = Optional.FromValue(command.Options);
+                        action.DefaultPermission = command.DefaultPermission;
+                    });
+
+                    Commands.Add(discord_backend_command);
+                }
+            }
+
+            if (CheckAnyAndNotNull(GuildCommandsDeleteList))
+            {
+                foreach (var cmdId in GuildCommandsDeleteList)
+                {
+                    try
+                    {
+                        await ApplicationCommandsExtension._client.DeleteGuildApplicationCommandAsync(GuildId, cmdId);
+                    }
+                    catch (NotFoundException)
+                    {
+                        ApplicationCommandsExtension._client.Logger.LogError($"Could not delete guild command {cmdId} in guild {GuildId}. Please clean up manually");
+                    }
+                }
+            }
+
+            return CheckAnyAndNotNull(Commands) ? Commands : null;
+        }
+
+        internal static bool CheckNotAnyOrNull(List<DiscordApplicationCommand> list)
+            => list == null || !list.Any();
+
+        internal static bool CheckNotAnyOrNull(List<ulong> list)
+            => list == null || !list.Any();
+
+        internal static bool CheckNotAnyOrNull(Dictionary<ulong, DiscordApplicationCommand> dict)
+            => dict == null || !dict.Any() || dict.Keys == null || !dict.Keys.Any();
+
+        internal static bool CheckAnyAndNotNull(List<DiscordApplicationCommand> list)
+            => list != null && list.Any();
+
+        internal static bool CheckAnyAndNotNull(List<ulong> list)
+            => list != null && list.Any();
+
+        internal static bool CheckAnyAndNotNull(Dictionary<ulong, DiscordApplicationCommand> dict)
+            => dict != null && dict.Any() && dict.Keys != null && dict.Keys.Any();
+
```

Comment:
omg I can do multi line comment. --- We could make an extension method for IEnumerable / IDictionary types for these same methods. Would make it reusable for other parts of DCS

Reply:
scratch that, not extension method -- utility method.

Reply:
Will do